mvrx-navigation/src/main/kotlin/com/airbnb/mvrx/navigation/navigationLifecycleAwareLazy.kt

```diff
@@ -0,0 +1,82 @@
+@file:Suppress("ClassName")
+
+package com.airbnb.mvrx.navigation
+
+import androidx.annotation.RestrictTo
+import androidx.annotation.VisibleForTesting
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.LifecycleOwner
+import java.io.Serializable
+
+private object UninitializedValue
+
+/**
+ * This was copied from SynchronizedLazyImpl but modified to automatically initialize in ON_CREATE.
+ */
+@RestrictTo(RestrictTo.Scope.LIBRARY)
+@SuppressWarnings("Detekt.ClassNaming")
+class navigationLifecycleAwareLazy<out T>(
+    owner: LifecycleOwner,
+    initializer: () -> T
+) : Lazy<T>, Serializable {
+    private var initializer: (() -> T)? = initializer
+
+    @Volatile
+    @SuppressWarnings("Detekt.VariableNaming")
+    private var _value: Any? = UninitializedValue
+
+    // final field is required to enable safe publication of constructed instance
+    private val lock = this
+
+    @VisibleForTesting
+    internal val lifecycleObserver: DefaultLifecycleObserver =
+        object : DefaultLifecycleObserver {
+            override fun onStart(owner: LifecycleOwner) {
+                super.onStart(owner)
+                try {
+                    if (!isInitialized()) value
+                    owner.lifecycle.removeObserver(this)
+                } catch (cause: IllegalStateException) {
+                    throw IllegalStateException(
+                        """
+                            During device re-configuration or launch after process death the navController is not accessible until onViewCreated.
+                             Try moving any ViewModel access to onViewCreated or later and use subscriptionLifecycleOwner for any subscriptions.
```

Comment:
Hmm, this is an interesting case. `viewModel.onEach` calls should be done from `onCreate`.
Is that not possible for nav graph scoped view models?

Also, from a semantics standpoint, `subscriptionLifecycleOwner` is not something most people who are using Mavericks will understand. I like that you provided resolution steps but it led to my concern above.

Reply:
> Hmm, this is an interesting case. viewModel.onEach calls should be done from onCreate.
Is that not possible for nav graph scoped view models?

The NavController is not available until the activity has re-attached the NavHost fragment after the Activity.onCreate has setContent and the FragmentManager has re-attached the fragments. So, while this will work when first creating the fragment, the NavController will not be accessible until the Fragments `onCreateView`.

> Also, from a semantics standpoint, subscriptionLifecycleOwner is not something most people who are using Mavericks will understand. I like that you provided resolution steps but it led to my concern above.

I don't think you can fix this case. Nav Graph ViewModels are limited by the accessibility of the BackStack entry and they are only guaranteed to be accessible via NavController in or after `onCreateView/onViewCreared`. We could just omit the last part and suggest moving ViewModel access to that lifecycle point.

```suggestion
                             Try moving any ViewModel access to `onViewCreated`.
```